src/test/java/org/opensearch/neuralsearch/highlight/SemanticHighlighterIT.java

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.highlight;
+
+import static org.opensearch.neuralsearch.util.TestUtils.TEST_DIMENSION;
+import static org.opensearch.neuralsearch.util.TestUtils.TEST_SPACE_TYPE;
+import static org.opensearch.neuralsearch.util.TestUtils.createRandomVector;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Locale;
+
+import org.junit.Before;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.MatchQueryBuilder;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+import org.opensearch.index.query.QueryStringQueryBuilder;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import com.google.common.primitives.Floats;
+import lombok.SneakyThrows;
+
+public class SemanticHighlighterIT extends BaseNeuralSearchIT {
```

Comment:
is it possible to combine some of the tests into a single method to minimize number of time model deployments? You're testing happy path for different query types, and each separate method required the same model to be deployed and undeployed. So essentially you can use same model id and test multiple queries with that model. Ofter when integ test are failing in CI during this model deployment step.

Reply:
Makes sense, updated IT to use shared resources.

Reply:
another point is concurrent execution of tests by test runner in CI, we had problems in the past with global  model_id that is shared between test methods

Reply:
The cleanup() will can be called in SemanticHighlighterIT after tests as well.

> another point is concurrent execution of tests by test runner in CI, we had problems in the past with global model_id that is shared between test methods

I don't see current CI failed due to the concurrent execution.

Reply:
Offline talked with @martin-gaievski, going to merge different query type test cases into one.